### ISSUE TITLE
Oracle - Unexpected whitespace before statement terminator '/'

### ISSUE THREAD

**Issue by nickbayley** [sqlfluff/sqlfluff#6842]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Since upgrading to v3.4.0 we are encountering a number of violations in our SQL due to `/` being seen as a `statement_terminator` instead of `binary_operator`. I believe this might be due to the recent addition of Oracle PL/SQL support.

### Expected Behaviour

```sql
select quantity / 100
from inventory;
```

The above SQL should lint with no errors.

### Observed Behaviour

```sql
select quantity / 100
from inventory;
```

Instead, we receive this violation:

```
L:   1 | P:  16 | LT01 | Unexpected whitespace before statement terminator '/'.
                       | [layout.spacing]
```

### How to reproduce

See example above.

### Dialect

```
oracle
```

### Version

```
sqlfluff, version 3.4.0
```

### Configuration

```
[sqlfluff]
dialect = oracle
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
